### PR TITLE
feat: support show all enable plugin list tab

### LINKED ISSUE #2299: 插件相关界面添加 "已启用" 分组选项，Insert new ButtonGroup component which with `All` , `Enabled` Child items in Plugin List View.
# Feature request

- Route > Route List > Create or Configure > `STEP 3 Plugin Config`
- Service > Create Service > `Step 2 Plugin`
- Plugin > Plugin List > Enable > `Global Plugin List`

## Please describe your feature

Insert new ButtonGroup(or Tabs) component which with `All` , `Enabled` child items. 

When child item is selected `Plugin List` shows the results by filter `all` or `enabled` plugins

界面上添加一个按钮组或者选项卡组，选项为 默认选中 `全部` 和`已启用`  ，快速区分`已经启用`的插件功能。

## Additional context

有没有计划做一个类似 `插件市场` 的功能? 或者在线添加一个插件。

### PR DESCRIPTION
**Pull request by Morakes**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [X] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #2299 

**Checklist:**

- [X] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: web/src/components/Plugin/PluginPage.tsx [hunk 1]
```diff
@@ -100,6 +100,30 @@ const PluginPage: React.FC<Props> = ({
     return index === self.indexOf(elem);
   });
 
+  const [selectedTab, setSelectedTab] = useState<string>('add');
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:103]:
```suggestion
  const [showEnablePlugin, setShowEnablePlugin] = useState(false);
```
is better?

**Comment by guoqqqi** [comment]
Hello @Morakes Can you help by posting a before and after picture of the change so that we can easily help review 😄

File: web/src/components/Plugin/PluginPage.tsx [hunk 2]
```diff
@@ -198,7 +222,13 @@ const PluginPage: React.FC<Props> = ({
                 pluginList.filter(
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:222]:
```suggestion
                (isEnablePlugin ? enablePluginList : pluginList).filter(
```
Actually, another good idea is that we have a variable for enablePluginList

**Comment by guoqqqi** [comment]
also need to resolve these conflicts 
image

**Comment by Morakes** [comment]
before
0f382affc8762307c1598287b3f1f9f

after
c6f9d8dde922a7edd81d96345ab3fd9
0ad59c335399ea9c3e8b0c83dab3c73

File: web/src/components/Plugin/PluginPage.tsx [hunk 3]
```diff
@@ -104,6 +104,34 @@ const PluginPage: React.FC<Props> = ({
...
+      key: 'enable',
+    },
+    {
+      title: formatMessage({ id: 'component.plugin.disable' }),
+      key: 'edit',
+    },
+  ];
+  const SwitchTab = () => (
+    <Tabs
+      defaultActiveKey={showEnablePlugin ? 'enable' : 'edit'}
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:120]:
```suggestion
      defaultActiveKey={showEnablePlugin ? 'enablePlugins' : 'allPlugins'}
```

File: web/src/components/Plugin/PluginPage.tsx [hunk 4]
```diff
@@ -104,6 +104,34 @@ const PluginPage: React.FC<Props> = ({
     return index === self.indexOf(elem);
   });
 
+  const [showEnablePlugin, setShowEnablePlugin] = useState<boolean>(true);
+  const tabsList = [
+    {
+      title: formatMessage({ id: 'component.plugin.enable' }),
+      key: 'enable',
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:111]:
```suggestion
      key: 'enablePlugins',
```

File: web/src/components/Plugin/PluginPage.tsx [hunk 5]
```diff
@@ -104,6 +104,34 @@ const PluginPage: React.FC<Props> = ({
...
+    },
+    {
+      title: formatMessage({ id: 'component.plugin.disable' }),
+      key: 'edit',
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:115]:
```suggestion
      key: 'allPlugins',
```

File: web/src/components/Plugin/PluginPage.tsx (hunk 3 above)
Comment (guoqqqi) [reply web/src/components/Plugin/PluginPage.tsx:120]: Not a good variable name

File: web/cypress/integration/route/create-route-with-cors-form.spec.js [hunk 6]
```diff
@@ -62,7 +64,7 @@ context('Create and delete route with cors form', () => {
     cy.get(selector.nodes_0_port).clear().type(data.port);
     cy.get(selector.nodes_0_weight).clear().type(data.weight);
     cy.contains('Next').click();
-
+    cy.wait(1000);
```
Comment (guoqqqi) [inline web/cypress/integration/route/create-route-with-cors-form.spec.js:67]: IMO, We refuse to abuse `wait`, it wastes a lot of our time, and we don't use it except as a last resort. Where we can use some assertions to ensure that the test is stable.

Comment (Morakes) [reply web/cypress/integration/route/create-route-with-cors-form.spec.js]: OK, I've fixed it, you can check it.

File: web/src/components/Plugin/PluginPage.tsx [hunk 7]
```diff
@@ -96,23 +97,32 @@ const PluginPage: React.FC<Props> = ({
...
     (item) => initialData[item.name] && !initialData[item.name].disable,
   );
+
   const openPluginType = openPluginList.map((item) => item.type);
   const newOpenPluginType = openPluginType.filter((elem, index, self) => {
     return index === self.indexOf(elem);
   });
 
-  const [showEnablePlugin, setShowEnablePlugin] = useState<boolean>(true);
+  const [showEnablePlugin, setShowEnablePlugin] = useState<boolean>(false);
```
Comment (guoqqqi) [inline web/src/components/Plugin/PluginPage.tsx:116]:
```suggestion
  const [showEnablePlugin, setShowEnablePlugin] = useState(false);
```
move this line to 71

**Comment by guoqqqi** [comment]
also cc @SkyeYoung PTAL

File: web/src/components/Plugin/PluginPage.tsx (hunk 7 above)
Comment (Morakes) [reply web/src/components/Plugin/PluginPage.tsx]: OK

**Review by SkyeYoung — APPROVED** [review]

File: web/src/components/Plugin/PluginPage.tsx [hunk 8]
```diff
@@ -104,6 +114,34 @@ const PluginPage: React.FC<Props> = ({
...
+
+  const SwitchTab = () => (
+    <Tabs
+      defaultActiveKey={showEnablePlugin ? 'enablePlugins' : 'allPlugins'}
+      onChange={(val: string) => {
+        if (val === 'enablePlugins') {
+          setShowEnablePlugin(true);
+        } else {
+          setShowEnablePlugin(false);
+        }
```
Comment (SkyeYoung) [inline web/src/components/Plugin/PluginPage.tsx:136]:
```suggestion
        setShowEnablePlugin(val === 'enablePlugins');
```

File: web/src/components/Plugin/PluginPage.tsx [hunk 9]
```diff
@@ -96,6 +98,14 @@ const PluginPage: React.FC<Props> = ({
...
   }, []);
+
+  useEffect(() => {
+    const openList = pluginList.filter(
+      (item) => initialData[item.name] && !initialData[item.name].disable,
+    );
+    setEnablePluginsList(openList);
+  }, [initialData]);
+
   const openPluginList = pluginList.filter(
```
Comment (SkyeYoung) [inline web/src/components/Plugin/PluginPage.tsx:109]: Can you help us improve this section, please? Like you've done here? 

https://github.com/Morakes/apisix-dashboard/blob/2877db87a5c3d8af1e187f012c406bd38f27f24a/web/src/components/Plugin/PluginPage.tsx#L102-L107

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

File: web/src/components/Plugin/PluginPage.tsx [hunk 10]
```diff
@@ -96,6 +98,14 @@ const PluginPage: React.FC<Props> = ({
...
     (item) => initialData[item.name] && !initialData[item.name].disable,
   );
```
Comment (SkyeYoung) [inline web/src/components/Plugin/PluginPage.tsx:111]: Cloud you help us improve this section, please? Like you've done here? 

https://github.com/Morakes/apisix-dashboard/blob/2877db87a5c3d8af1e187f012c406bd38f27f24a/web/src/components/Plugin/PluginPage.tsx#L102-L107

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

File: web/src/components/Plugin/PluginPage.tsx (hunk 10 above)
Comment (Morakes) [reply web/src/components/Plugin/PluginPage.tsx:111]: Ok, I will improve it

**Review by Baoyuantop — COMMENTED** [review]
There is an issue that needs to be verified and fixed:
1. Go to the Plugin List page and add Enable
2. Enable `basic auth` and `key auth` plugin
3. Switch tabs and data are as expected
4. Click the Plugin menu and go to the plugin list page
5. Click Enable button and click Enable tab
6. Delete the last plugin and find that deleted both

**Comment by Baoyuantop** [comment]
> There is an issue that needs to be verified and fixed:
> 
> 1. Go to the Plugin List page and add Enable
> 2. Enable `basic auth` and `key auth` plugin
> 3. Switch tabs and data are as expected
> 4. Click the Plugin menu and go to the plugin list page
> 5. Click Enable button and click Enable tab
> 6. Delete the last plugin and find that deleted both

Oh sorry I find this issue in https://github.com/apache/apisix-dashboard/issues/2592, please ignore this comment

**Review by Baoyuantop — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

File: web/src/components/Plugin/PluginPage.tsx [hunk 11]
```diff
@@ -66,6 +66,8 @@ const PluginPage: React.FC<Props> = ({
   const { formatMessage } = useIntl();
   const [form] = Form.useForm();
   const [pluginList, setPluginList] = useState<PluginComponent.Meta[]>([]);
+  const [enablePluginsList, setEnablePluginsList] = useState<PluginComponent.Meta[]>([]);
```
Comment (oil-oil) [inline web/src/components/Plugin/PluginPage.tsx:69]: enablePluginsList can be obtained from the `pluginList` after calculation, no need to define a new state.

**Pull request merged**